### PR TITLE
NF: Output BOLD HMC transforms and reference volume

### LINKED ISSUE #2803: Output motion transforms in BIDS Derivatives
Affine transforms from each original BOLD volume to the BOLD reference volume should be stored in the derivatives.

For now let's use a simple format that roughly corresponds to an array such that:

```Python
# affine : list of 4x4 matrices
# all_params : n x 12 matrix
affine[i][:3] = all_params[i, :].reshape(3, 4)
```

Part of #2207.

### PR DESCRIPTION
**Pull request by effigies**
Closes #2803. The format is up for debate, but this is what we use internally, and it's pretty reasonable.

cc @lestropie

### CODE REVIEW

**Comment by effigies** [comment]
Here's an example (link probably won't work for long):
[tmp/ds054/derivatives/fmriprep/sub-100185/func/sub-100185_task-machinegame_run-01_from-scanner_to-boldref_mode-image_xfm.txt](https://output.circle-artifacts.com/output/job/4f15eda2-e959-4ce6-b934-2b802eeac469/artifacts/0/tmp/ds054/derivatives/fmriprep/sub-100185/func/sub-100185_task-machinegame_run-01_from-scanner_to-boldref_mode-image_xfm.txt)

```
#Insight Transform File V1.0
#Transform 0
Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 1 -5.63032e-05 9.56292e-05 5.64124e-05 0.999999 -0.00132235 -9.55647e-05 0.00132235 0.999999 -0.00951554 0.154872 -0.0395518
FixedParameters: 0 0 0
#Transform 1
Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 1 0.000103325 0.000299206 -0.000103188 1 -0.000229664 -0.000299254 0.000229602 1 -0.0127712 0.0818696 0.0471371
FixedParameters: 0 0 0
#Transform 2
Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 1 9.66473e-05 -0.000163583 -9.63369e-05 0.999998 0.00226405 0.000163766 -0.00226404 0.999998 -0.0114369 0.0454862 0.191359
FixedParameters: 0 0 0
#Transform 3
Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 1 -1.0613e-05 1.80126e-05 1.0622e-05 1 -0.000249124 -1.80073e-05 0.000249124 1 -0.00918892 0.15731 -0.0512331
FixedParameters: 0 0 0
#Transform 4
Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 0.999999 0.00030273 0.000642284 -0.000303562 0.999999 0.000647455 -0.000641891 -0.000647844 0.999999 -0.00834608 0.0187707 0.0370819
FixedParameters: 0 0 0
```

File: fmriprep/workflows/bold/base.py [hunk 1]
```diff
@@ -369,6 +369,7 @@ def init_func_preproc_wf(bold_file, has_fieldmap=False):
                 "bold_t1_ref",
                 "bold2anat_xfm",
                 "anat2bold_xfm",
+                "hmc_xforms",
```
Comment (mgxd) [inline fmriprep/workflows/bold/base.py:372]: Should add this to the docstring outputs

File: fmriprep/workflows/bold/outputs.py [hunk 2]
```diff
@@ -220,6 +220,7 @@ def init_func_derivatives_wf(
                 't2star_std',
                 'bold2anat_xfm',
                 'anat2bold_xfm',
+                'hmc_xforms',
```
Comment (mgxd) [inline fmriprep/workflows/bold/outputs.py:223]: And this one to the docstring inputs

**Review by mgxd — APPROVED** [review]
LGTM

File: fmriprep/workflows/bold/outputs.py (hunk 2 above)
Comment (effigies) [reply fmriprep/workflows/bold/outputs.py:223]: This function doesn't have docstrings for input/output nodes.

**Pull request merged**